src/dbnode/encoding/proto/custom_marshal_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package proto
+
+import (
+	"testing"
+
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCustomMarshal(t *testing.T) {
+	testCases := []struct {
+		message *dynamic.Message
+	}{
+		{
+			message: newVL(0, 0, 0, nil, nil),
+		},
+		{
+			message: newVL(1, 1, 1, []byte("some-delivery-id"), nil),
+		},
+		{
+			message: newVL(2, 2, 2, []byte("some-delivery-id-2"), map[string]string{"key": "val"}),
+		},
+	}
+
+	marshaler := newCustomMarshaler()
+	for _, tc := range testCases {
+		marshaler.reset()
+		customMarshalVL(t, marshaler, tc.message)
+
+		unmarshalM := dynamic.NewMessage(testVLSchema)
+		require.NoError(t, unmarshalM.Unmarshal(marshaler.bytes()))
+
+		require.True(t, dynamic.Equal(tc.message, unmarshalM))
```

Comment:
So you can run individual tests, etc it's best practice to wrap this in a `t.Run(...)`:
```go
for i, tc := range testCases {
  t.Run(fmt.Sprintf("my_test_%d", i), func (t *testing.T) {
    // test case innards
  })
}
```

Also helps if you need defer, etc in the body.